### DISCUSSION TITLE
Phoeting

### DISCUSSION THREAD

**Discussion by MohammadReza0GH** [phoebe-project/phoebe2#865] [Questions]
hi, i was wandering how in this ( https://arxiv.org/abs/2307.13482 
[2307.13482_2.pdf](https://github.com/phoebe-project/phoebe2/files/15249886/2307.13482_2.pdf)
) paper they did fitting?! there is no parameters used for primary. but they use semidetached constraint. if i want to try fitting like the system they use what must i do?? for requiv1? and for teff1?? i got errors in every try i did. requiv_max error and ld_coeff errors and atm error that they use ck2004 but i cant, and soooo many errors. so in the end im tired and confused, help me. :((((((((

**Comment by Raindogjones** [comment]
See the tutorial on semi-detached systems [here](http://phoebe-project.org/docs/2.4/tutorials/requiv_crit_semidetached)

**Reply by MohammadReza0GH** [reply]
i did already, in tutorial we change atm to blackbody, but they use ck2004 in paper.

**Reply by Raindogjones** [reply]
For the limb-darkening coefficient, yes, not for the atmosphere.
b.set_value_all('ld_mode','lookup')
b.set_value_all('ld_coeffs_source','ck2004')
b.set_value_all('ld_func','logarithmic')

See the tutorial on LD coefficients [here](http://phoebe-project.org/docs/latest/tutorials/limb_darkening)

**Reply by MohammadReza0GH** [reply]
yes i did that too, but code doesnt start to run. i wait for 1h but it didnt start to run.
![Screenshot (31)](https://github.com/phoebe-project/phoebe2/assets/167889059/9e14d5a7-9fc2-4fbb-be29-dcc0ce231076)

**Reply by Raindogjones** [reply]
Ok, unfortunately, without more information or a minimal working example of where/how it fails, I cannot help. Sorry.

**Reply by MohammadReza0GH** [reply]
[DATA010.txt](https://github.com/phoebe-project/phoebe2/files/15250549/DATA010.txt)

this is data that i use, its on phase but we have period = 0.1725 and we can convert it to time.
-semidetached, eclipse_method = only_horizon, no information of requiv1 but we know its NS(pulsar)
can you please try an estimator or optimizer to fit? they didnt work for me, i got fits like this:

![okokokokoko](https://github.com/phoebe-project/phoebe2/assets/167889059/62d8656b-de73-43b0-a02c-941035ca452f)

**Reply by Raindogjones** [reply]
If you want us to try to reproduce (or diagnose) your error, please send a minimal working example of code.

**Reply by MohammadReza0GH** [reply]
b.add_constraint('semidetached', 'secondary')
b.set_value_all('period@binary', value = 0.1725)
b.set_value('sma@binary', value=0.219)
b.set_value('q', value=0.05)
b.set_value('requiv@primary', value=0.00014)
b.set_value('eclipse_method', value='only_horizon')
b.set_value('ld_func_bol@secondary', value='logarithmic')
b.set_value('gravb_bol@secondary', value=0.08)
b.add_dataset('lc', times=times, fluxes = lc[:,1]*10**-17, pblum_mode='absolute', passband='SDSS:g')
b['ld_coeffs_source_bol@secondary']='ck2004'
b['ld_mode_bol@primary']='manual'
#b['atm@primary'] = 'blackbody'

b.add_compute(compute='lc_fit')
b.set_value('distance@system', value=1, unit='pc')
b.set_value('eclipse_method',compute='lc_fit', value='only_horizon')
b.add_distribution({'q@binary': phoebe.uniform(0.04,0.12),
                    'incl@binary': phoebe.uniform(50,90),
                    'teff@secondary': phoebe.uniform(3500,7000),
                    'requiv@secondary': phoebe.gaussian(0.27),
                    'Av@system' : phoebe.uniform(0.7,0.8),
                    'distance@system': phoebe.uniform(1664,7766)}, distribution ='NNN', overwrite_all=True)

b['passband']='SDSS:g'
b.run_compute(compute='lc_fit', sample_from='NNN', sample_num=60,overwrite=True)
b.add_solver('sampler.emcee', solver='mcmc',compute='lc_fit', init_from='NNN')

b.run_solver('mcmc', solution="lc44")

**Reply by Raindogjones** [reply]
In that code you don't ever load a binary nor the datafile, so this is not a minimal working example.

Above I explained that you probably need to set the limb-darkening but you only set the bolometric limb-darkening (used for irradiation).  
The secondary atmosphere probably also goes out of bounds if you aren't setting it to blackbody anywhere.
You are trying to sample/set the secondary requiv but this is automatically set by the semi-detached constraint, so cannot be done.

**Reply by kecnry** [reply]
It looks like its actually the `run_compute` with `sample_num=60` that is stalling.  I'm guessing that is because the initial samples are all getting rejected and are unable to compute.  My suggestion would be to check your sampling distribution carefully to make sure it is valid.

**Reply by MohammadReza0GH** [reply]
so if i dont set secondary to be constraint i got overflowing error! this in my main problem, how they did it in paper that i mentioned?
![Screenshot (32)](https://github.com/phoebe-project/phoebe2/assets/167889059/25ab2161-7752-4c3d-bafa-11e57c7a55a0)
![Screenshot (33)](https://github.com/phoebe-project/phoebe2/assets/167889059/c8499db1-5c20-466b-b615-c06c79907dd8)

**Reply by kecnry** [reply]
To see the actual errors that are causing the samples to be immediately rejected, you can always manually sample your distribution and adopt the face-values (with [sample_distribution_collection](http://phoebe-project.org/docs/2.4/api/phoebe.frontend.bundle.Bundle.sample_distribution_collection)`('NNN', set_value=True)` and call `run_compute()` on those.

You can also reach out to the authors of the paper if you're unable to reproduce their results.  Sometimes some details are not included in the text that can be important to reproduce.

**Reply by MohammadReza0GH** [reply]
OK, thanks a lot kyle.
